### DISCUSSION TITLE
[Typography]: New component discussion

### DISCUSSION THREAD

**Discussion by webvs2** [element-plus/element-plus#14056] [General]
对于该组件，定位主要是对文章基础快速排版，和各项如图片，表格等内容的配合展示。
那么展示方面基本的分为，主体，段落，标题，文本，填充内容。其中文本可有数学，代码，emoji（这些做成大众的样式来快速呈现）。填充内容在于图片或者表格相较于段落是如何的一个内容展示，是类似内嵌的那种吗。
在程序方面，我认为一个较大作用在于服务端渲染，对于这些静态内容更好的SEO和渲染速度，要考虑下。

**Comment by webvs2** [comment]
@YunYouJun

**Reply by YunYouJun** [reply]
看起来你期待的更像是一个 Markdown CSS 与 [vitesse](https://github.com/antfu/vitesse) 中 markdown 的渲染方式 [unplugin-vue-markdown](https://github.com/unplugin/unplugin-vue-markdown)。

数学公式需要额外的第三方包，如 KaTeX，代码高亮也相对复杂，需要第三方库实现，如 Highlight/Prism/Shiki 等。
这些看起来不是，也无法由组件服务端渲染提供。服务端渲染需要预编译时，无法通过客户端组件实现，而需要用户引入第三方包并配置（需要加到打包流程里）。
